datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -581,7 +597,20 @@ macro_rules! get_data_page_statistics {
                 )),
                 Some(DataType::Int32) => Ok(Arc::new(Int32Array::from_iter([<$stat_type_prefix Int32DataPageStatsIterator>]::new($iterator).flatten()))),
                 Some(DataType::Int64) => Ok(Arc::new(Int64Array::from_iter([<$stat_type_prefix Int64DataPageStatsIterator>]::new($iterator).flatten()))),
-                _ => unimplemented!()
+                Some(DataType::Float16) => Ok(Arc::new(
+                    Float16Array::from_iter(
+                        [<$stat_type_prefix Float32DataPageStatsIterator>]::new($iterator)
+                            .map(|x| {
+                                x.into_iter().filter_map(|x| {
+                                    x.and_then(|x| Some(f16::from_f32(x)))
```

Comment:
@tshauck 
I think you can mimic the approach [here](https://github.com/apache/datafusion/blob/766ee0480f73183a1ebc27b8243cfacbd7903c7c/datafusion/core/src/datasource/physical_plan/parquet/statistics.rs#L336-L338)? This might also fix the failing test.